File: app/src/main/java/knf/kuma/tv/anime/ChapterPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.tv.cards.ChapterCardView;

public class ChapterPresenter extends Presenter {

    public ChapterPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new ChapterCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((ChapterCardView) viewHolder.view).bind((AnimeObject.WebInfo.AnimeChapter) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/ChapterPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import androidx.lifecycle.LifecycleCoroutineScope
import knf.kuma.pojos.av1.ChapterWID
import knf.kuma.tv.cards.ChapterCardView

class ChapterPresenter(val scope: LifecycleCoroutineScope) : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(ChapterCardView(parent.context, scope))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any?) {
        if (item == null) return
        (viewHolder.view as ChapterCardView).bind(item as ChapterWID)
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {
        (viewHolder.view as ChapterCardView).unbind()
    }
}
